### PR TITLE
[subset] when dropping hints, clear hint related fields in maxp.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: src/hb-ot-maxp-table.hh [hunk 1]
```diff
@@ -73,19 +73,56 @@ struct maxp
...
+      maxp_prime->version_1_data.maxSizeOfInstructions.set (0);
+    }
+  }
+
   protected:
   FixedVersion<>version;		/* Version of the maxp table (0.5 or 1.0),
 					 * 0x00005000u or 0x00010000u. */
   HBUINT16	numGlyphs;		/* The number of glyphs in the font. */
+
+  struct
```
Comment (behdad) [inline src/hb-ot-maxp-table.hh]: Please move this out to its own struct and add sanitize() and other methods to it, and reference from maxp table. See how it's done in `hb-ot-post-table.hh`. Thanks

Comment (garretrieger) [reply src/hb-ot-maxp-table.hh]: Done.

File: src/hb-ot-maxp-table.hh [hunk 2]
```diff
@@ -39,6 +39,36 @@ namespace OT {
...
+    TRACE_SANITIZE (this);
+    return_trace (c->check_struct (this));
+  }
+
+  HBUINT16 maxPoints;             /* Maximum points in a non-composite glyph. */
+  HBUINT16 maxContours;           /* Maximum contours in a non-composite glyph. */
+  HBUINT16 maxCompositePoints;    /* Maximum points in a composite glyph. */
+  HBUINT16 maxCompositeContours;  /* Maximum contours in a composite glyph. */
+  HBUINT16 maxZones;              /* 1 if instructions do not use the twilight zone (Z0),
+                                     or 2 if instructions do use Z0; should be set to 2 in
```
Comment (behdad) [inline src/hb-ot-maxp-table.hh]: Please add '*' to the start of comment continuation lines.

File: src/hb-ot-maxp-table.hh [hunk 3]
```diff
@@ -39,6 +39,36 @@ namespace OT {
...
+  HBUINT16 maxStorage;            /* Number of Storage Area locations. */
+  HBUINT16 maxFunctionDefs;       /* Number of FDEFs, equal to the highest function number + 1. */
+  HBUINT16 maxInstructionDefs;    /* Number of IDEFs. */
+  HBUINT16 maxStackElements;      /* Maximum stack depth. (This includes Font and CVT
+                                     Programs, as well as the instructions for each glyph.) */
+  HBUINT16 maxSizeOfInstructions; /* Maximum byte count for glyph instructions. */
+  HBUINT16 maxComponentElements;  /* Maximum number of components referenced at
+                                     "top level" for any composite glyph. */
+  HBUINT16 maxComponentDepth;     /* Maximum levels of recursion; 1 for simple components. */
+
```
Comment (behdad) [inline src/hb-ot-maxp-table.hh]: Just by convention we always put the public: before the DEFINE_..., and with no empty line before it.

File: src/hb-ot-maxp-table.hh [hunk 4]
```diff
@@ -83,46 +119,27 @@ struct maxp
 
   static inline void drop_hint_fields (hb_subset_plan_t *plan, OT::maxp *maxp_prime)
   {
+    maxpV1Tail &v1 = StructAfter<maxpV1Tail> (*maxp_prime);
```
Comment (behdad) [inline src/hb-ot-maxp-table.hh]: This should go after the version check (inside the block).

File: src/hb-ot-maxp-table.hh [hunk 5]
```diff
@@ -83,46 +119,27 @@ struct maxp
...
-    HBUINT16 maxFunctionDefs;       /* Number of FDEFs, equal to the highest function number + 1. */
-    HBUINT16 maxInstructionDefs;    /* Number of IDEFs. */
-    HBUINT16 maxStackElements;      /* Maximum stack depth. (This includes Font and CVT
-                                       Programs, as well as the instructions for each glyph.) */
-    HBUINT16 maxSizeOfInstructions; /* Maximum byte count for glyph instructions. */
-    HBUINT16 maxComponentElements;  /* Maximum number of components referenced at
-                                       "top level" for any composite glyph. */
-    HBUINT16 maxComponentDepth;     /* Maximum levels of recursion; 1 for simple components. */
-  } version_1_data;
+/*maxpV1Tail v1Tail; */
```
Comment (behdad) [inline src/hb-ot-maxp-table.hh]: Please change to:
/*maxpV1Tail v1Tail[VAR]; */

File: src/hb-ot-maxp-table.hh (hunk 2 above)
Comment (garretrieger) [reply src/hb-ot-maxp-table.hh]: Done.

File: src/hb-ot-maxp-table.hh (hunk 3 above)
Comment (garretrieger) [reply src/hb-ot-maxp-table.hh]: Done.

File: src/hb-ot-maxp-table.hh (hunk 4 above)
Comment (garretrieger) [reply src/hb-ot-maxp-table.hh]: Done.

File: src/hb-ot-maxp-table.hh (hunk 5 above)
Comment (garretrieger) [reply src/hb-ot-maxp-table.hh]: Done.

**Pull request merged**